### ISSUE TITLE
Using of nested structure (dot syntax fields) breaks single_table_references

### ISSUE THREAD

**Issue by KulykDmytro** [sqlfluff/sqlfluff#6250]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When single table statement contains complex columns and `single_table_references` set to `consistent` in result given option is not honored
```sql
select
    a,
    b,
    c.struct.val
from sch.t1
```

### Expected Behaviour

should violate on inconsistency of `c.struct.val` only cause `a` and `b` are referenced normally due to single table query

### Observed Behaviour

following violations are caught:
```verilog
L:   2 | P:   5 | RF03 | Unqualified reference 'a' found in single table select. [references.consistent]
L:   3 | P:   5 | RF03 | Unqualified reference 'b' found in single table select. [references.consistent]
L:   4 | P:   5 | RF03 | Qualified reference 'c.struct.val' found in single table select which is inconsistent with previous references. [references.consistent]
L:   4 | P:   7 | RF01 | Reference 'c.struct.val' refers to table/view not found in the FROM clause or found in ancestor statement. [references.from]
```
however this one works fine
```sql
select
    a,
    b,
    c
from sch.t1
```
```sql
select
    t.a,
    t.b,
    t.c
from sch.t1 as t
```

### How to reproduce

-

### Dialect

athena

### Version

3.2.0

### Configuration

[sqlfluff]
templater = jinja
output_line_length = 180
dialect = athena
large_file_skip_char_limit = 0
large_file_skip_byte_limit = 0
max_line_length = 120
warn_unused_ignores = true
processes = 4

[sqlfluff:rules:references.consistent]
single_table_references = consistent

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by KulykDmytro** [comment]
i beleive this is caused by #6067 when behaviour became more "greedy"
so need to split these cases on:
* ones where you can reuse column from ancestor:
  * `where val in (select ...)`
* ones when not:
  * `from (select from ...)`

**Comment by KulykDmytro** [comment]
patrially fixed by #6231

**Comment by Rishi943** [comment]
Working on a fix for this - will open a PR shortly.
